packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmMapInternal.kt

```diff
@@ -80,6 +81,8 @@ internal abstract class ManagedRealmMap<K, V>(
  */
 internal interface MapOperator<K, V> : CollectionOperator<V, RealmMapPointer> {
 
+    // Modification counter used to detect concurrent writes from the iterator
+    var modCount: Int
```

Comment:
I suspect we need to make this field volatile to avoid threading issues on JVM... not sure if that is possible in common code, alternatively make it atomic? 

How is Kotlin solving this in the std library?